sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveStrategies.scala

```diff
@@ -243,7 +243,8 @@ private[hive] trait HiveStrategies {
 
   object HiveScripts extends Strategy {
     def apply(plan: LogicalPlan): Seq[SparkPlan] = plan match {
-      case ScriptTransformation(input, script, output, child, ioschema) =>
+      case ScriptTransformation(input, script, output, child, ioschema)
+        if ioschema.inputSerdeClass.nonEmpty || ioschema.outputSerdeClass.nonEmpty =>
```

Comment:
Why do we need this condition `if ioschema.inputSerdeClass.nonEmpty || ioschema.outputSerdeClass.nonEmpty` now?

Reply:
> Why do we need this condition `if ioschema.inputSerdeClass.nonEmpty || ioschema.outputSerdeClass.nonEmpty` now?

Removed and see https://github.com/apache/spark/pull/29085#discussion_r456193004